gdx/src/com/badlogic/gdx/graphics/g2d/SpriteBatch.java

```diff
@@ -86,8 +95,16 @@ public SpriteBatch (int size) {
 	 * The defaultShader specifies the shader to use. Note that the names for uniforms for this default shader are different than
 	 * the ones expect for shaders set with {@link #setShader(ShaderProgram)}. See {@link #createDefaultShader()}.
 	 * @param size The max number of sprites in a single batch. Max of 5460.
-	 * @param defaultShader The default shader to use. This is not owned by the SpriteBatch and must be disposed separately. */
-	public SpriteBatch (int size, ShaderProgram defaultShader) {
+	 * @param defaultShader The default shader to use. This is not owned by the SpriteBatch and must be disposed separately.
+	 * @param extraAttributes Additional vertex attributes that this SpriteBatch will support beyond the required ones (Position,
+	 *           ColorPacked, TextureCoordinates), in the order they will be expected by the Mesh. If any draw method besides
+	 *           {@link #draw (Texture, float[], int, int)} is used, the extra vertex attributes will contain unexpected data that
+	 *           may be left over from a previous flush. If any extra vertex attributes are provided, this SpriteBatch can not be
+	 *           used for drawing standard objects that have their own {@code draw(Batch batch)} methods, such as
+	 *           {@linkplain Sprite}, {@linkplain BitmapFont}, and {@linkplain com.badlogic.gdx.scenes.scene2d.Stage Stage}. The
+	 *           {@linkplain ExpandableSprite} class can be used as a prototype for creating objects that can be drawn with the
+	 *           extra attributes. */
+	public SpriteBatch (int size, ShaderProgram defaultShader, VertexAttribute... extraAttributes) {

```

Comment:
As far as I know, varargs don't generate garbage. They generate an array which you may or may not hang onto with a member reference. In this case I don't hang onto it, but the user could pass in an array to begin with and keep the reference, as I do in the test.

However I was generating garbage for all the other constructors by passing `new VertexAttribute[0]`. I will put `(VertexAttribute[])null` instead.